src/main/java/core/basesyntax/utils/transaction/handlers/impl/BalanceOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.utils.transaction.handlers.impl;
+
+import core.basesyntax.storage.impl.StorageImpl;
+import core.basesyntax.utils.transaction.FruitTransaction;
+import core.basesyntax.utils.transaction.handlers.OperationHandler;
+
+public class BalanceOperation implements OperationHandler {
+    public void perform(FruitTransaction transaction, StorageImpl storage) {
```

Comment:
The `perform` method should not directly depend on `StorageImpl`. Instead, it should depend on an interface (e.g., `Storage`) to follow the Dependency Inversion Principle, one of the SOLID principles.